hail/python/hail/nd/nd.py

```diff
@@ -443,13 +442,15 @@ def vstack(arrs):
            [3],
            [4]], dtype=int32)
     """
-
-    if arrs[0].ndim == 1:
-        arrays = [x._broadcast(2) for x in arrs]
+    if isinstance(arrs, list):
```

Comment:
I think we can avoid this. If we change the type check from `tsequence_of(tndarray)` to `expr_array(expr_ndarray)` (or something like that), then all python arrays passed into this method will automatically be converted into hail arrays, and we won't need any branching.